### PR TITLE
[GP-12] support ANS search with partial APT suffix

### PR DESCRIPTION
**Pull request by 0xaptosj**
Let's say an user's ANS name is `name.apt`, previous we can only search it by `name` or `name.apt`, this PR let user search by `name`, `name.`, `name.a`, `name.ap`, `name.apt`

### CODE REVIEW

**Comment by 0xaptosj** [comment]
should i move the `yarn.lock` update to a separate PR?

File: src/api/hooks/useGetANS.ts [hunk 1]
```diff
@@ -54,11 +54,21 @@ function getFetchAddressUrl(network: NetworkName, name: string) {
   return `https://www.aptosnames.com/api/${network}/v1/address/${name}`;
 }
 
+function truncateAptSuffix(name: string): string {
```
Comment (jillxuu) [inline src/api/hooks/useGetANS.ts:57]: thanks for working on this issue! wondering can we use regex instead to find all the matching scenarios where "cathy.", "cathy.a", "cathy.ap", "cathy.apt" are all qualified and can be translated to "cathy"? 

some regex like `const regex = /^([a-zA-Z0-9-]+)(\.apt|\.ap|\.a|\.?)$/;`

Comment (0xaptosj) [reply src/api/hooks/useGetANS.ts:57]: i feel like if we introduce more characters to ANS in the future, the regex will be more and more complex? generally i tried to avoid using regex because of it's difficult to read or test. thoughts?

Comment (jillxuu) [reply src/api/hooks/useGetANS.ts:57]: in the forseeable future, we have primary names which is like "cathy.apt" and subdomains, which is like  "cathy.sub.apt". right now we only display primary names, and regex won't get complex as long as it fulfills the requirement "Name should have between 3 to 63 characters and contain only: lowercase (a-z), numbers (0-9), hyphen (-). A name may not start or end with a hyphen", which ANS uses regex to check against input string's validity. 

let me know if you have a strong preference here! we can discuss async as well

Comment (0xmaayan) [reply src/api/hooks/useGetANS.ts:57]: This is the regex we widely use 

 /^[a-z\d][a-z\d-]{1,61}[a-z\d]$/;

I think it is somewhere is the ANS codebase, @angieyth might be nice to export it as a util function from the sdk - something like isNameValid

File: yarn.lock [hunk 2]
```diff
@@ -38,13 +38,14 @@
     tslib "^2.3.0"
     zen-observable-ts "^1.2.5"
 
-"@aptos-labs/aptos-names-connector@^0.0.1":
-  version "0.0.1"
-  resolved "https://registry.yarnpkg.com/@aptos-labs/aptos-names-connector/-/aptos-names-connector-0.0.1.tgz#51f5ea6f6fc80b679ed4b03a29f35623231bcd25"
-  integrity sha512-HTVqbnsaKyfaqGdf7fC9dJsj8hr9aLS4mB5XUb8KxN+o/KKW4Ui0KpoPeNoLasCC+CZAcQohCvcPWJeRd5UNsg==
+"@aptos-labs/aptos-names-connector@^0.0.2":
```
Comment (0xmaayan) [inline yarn.lock:41]: Why these changes?

**Review by 0xmaayan — CHANGES_REQUESTED** [review]
Highly recommend adding some unit tests to test these changes

File: yarn.lock (hunk 2 above)
Comment (0xaptosj) [reply yarn.lock:41]: last [PR](https://github.com/aptos-labs/explorer/pull/440) only updated `package.json` but not `yarn.lock`

**Comment by 0xaptosj** [comment]
> Highly recommend adding some unit tests to test these changes

do we have any ts repo with unit test i can refer to? im new to frontend

**Comment by 0xmaayan** [comment]
> > Highly recommend adding some unit tests to test these changes
> 
> 
> 
> do we have any ts repo with unit test i can refer to? im new to frontend 🙂️

Can take a look here https://github.com/aptos-labs/governance/blob/main/src/__tests__/utils.test.tsx

**Review by 0xmaayan — APPROVED** [review]
